packages/compose/src/hooks/use-async-list/index.ts

```diff
@@ -50,11 +63,14 @@ function listReducer( state, action ) {
  * React hook returns an array which items get asynchronously appended from a source array.
  * This behavior is useful if we want to render a list of items asynchronously for performance reasons.
  *
- * @param {Array} list Source array.
- * @return {Array} Async array.
+ * @param list Source array.
+ * @return Async array.
  */
-function useAsyncList( list ) {
-	const [ current, dispatch ] = useReducer( listReducer, [] );
+function useAsyncList< T >( list: T[] ): T[] {
+	const [ current, dispatch ] = useReducer(
+		listReducer as Reducer< T[], ResetAction< T > | AppendAction< T > >,
+		[] as T[]
+	);
```

Comment:
Remember the Redux ducks pattern we discussed few days ago? We could use something similar in this hook:
```ts
const [ current, setCurrent ] = useState< T[] >( [] );

setCurrent( firstItems ); // replacement for reset action
setCurrent( state => [ ...state, item ] ); // replacement for append action
```
That saves a lot of effort typing the reducer and the actions. Many lines of code can be deleted.